File: PATTERNS/Pattern49/Pattern49.py

def isPrime(n):
    for i in range(2,n):
        if n % i == 0:
            return False
    return True

def prime(count):
    i = 1
    while(True):
        i += 1
        if isPrime(i):
            if count == 1:
                return i
            count -= 1

n = int(input("Enter the number of rows:"))

count = 1
        
for i in range(n+1):
    print(" " * (n-i), end="")
    for j in range(i):
        print(prime(count),end = " ")
        count += 1
    print()

File: PATTERNS/Pattern49/Pattern49.cpp
#include<iostream>
#include <math.h>
using namespace std;

int isPrime(int num) {
    if (num <= 1) {
        return 0;
    }
    
    for (int i = 2; i * i <= num; i++) {
        if (num % i == 0) {
            return 0;
        }
    }
    
    return 1;
}

void printLineOfPrimes(int offset, int lineDigits, int max) {
    if (lineDigits >= max) {
        return;
    }
    
    int primesInThisLine = 0;

    int space = max - lineDigits;

    for(int i = 0; i < space ; i++){
        cout << "  ";
    }

    while (primesInThisLine < lineDigits) {
        if (isPrime(offset)) {
            cout << offset << "  ";
            primesInThisLine++;
        }
        
        offset++;
    }
    
    cout << endl;
    printLineOfPrimes(offset, ++lineDigits, max);
}

int main() {
    int n;
    printf("Enter the number of lines N : ");
    cin >> n;
    
    printLineOfPrimes(1, 1, n + 1);
    
    return 0;
}